app/deployments/models.py

```diff
@@ -264,7 +264,10 @@ def group_timeseries_by_constraint(self):
         groups = defaultdict(list)
 
         for ts in self.timeseries_set.filter(end_time=None):
-            groups[tuple(ts.constraints.items())].append(ts)
+            try:
+                groups[tuple(ts.constraints.items())].append(ts)
+            except AttributeError as e:
+                logger.error(f"Unable to set constraint for timeseries {ts} due to {e}")
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use % formatting in logging functions and pass the % parameters as arguments](https://app.codacy.com/manual/gmri/buoy_barn/pullRequest?prid=4987011)